pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -345,7 +412,138 @@ func ValidateServiceInstanceReferencesUpdate(new *sc.ServiceInstance, old *sc.Se
 	return allErrs
 }
 
+func validateObjectReferences(spec *sc.ServiceInstanceSpec, fldPath *field.Path) field.ErrorList {
+	var errMsg string
+	allErrs := field.ErrorList{}
+
+	if spec.ClusterServiceClassRef != nil && spec.ServiceClassRef != nil {
+		errMsg = "ClusterServiceClassRef and ServiceClassRef should never be set simultaneously"
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("clusterServiceClassRef"), errMsg))
```

Comment:
Why is this a forbidden? I thought we were using invalid when two exclusive fields were set? Forbidden implies that you just can't set it, no? 🤔

Reply:
I recall mulling over exactly what type of errors these should be, I'll try to catch up with you in slack to discuss and update here with a decision.